x-pack/plugins/apm/public/components/alerting/AlertingFlyout/index.tsx

```diff
@@ -3,29 +3,38 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-import React from 'react';
+import React, { useMemo } from 'react';
+import { useKibana } from '../../../../../../../src/plugins/kibana_react/public';
 import { AlertType } from '../../../../common/alert_types';
-import { AlertAdd } from '../../../../../triggers_actions_ui/public';
-
-type AlertAddProps = React.ComponentProps<typeof AlertAdd>;
+import { TriggersAndActionsUIPublicPluginStart } from '../../../../../triggers_actions_ui/public';
 
 interface Props {
-  addFlyoutVisible: AlertAddProps['addFlyoutVisible'];
-  setAddFlyoutVisibility: AlertAddProps['setAddFlyoutVisibility'];
+  addFlyoutVisible: boolean;
+  setAddFlyoutVisibility: React.Dispatch<React.SetStateAction<boolean>>;
   alertType: AlertType | null;
 }
 
+interface KibanaDeps {
+  triggersActionsUi: TriggersAndActionsUIPublicPluginStart;
+}
+
 export function AlertingFlyout(props: Props) {
   const { addFlyoutVisible, setAddFlyoutVisibility, alertType } = props;
-  return (
-    alertType && (
-      <AlertAdd
-        addFlyoutVisible={addFlyoutVisible}
-        setAddFlyoutVisibility={setAddFlyoutVisibility}
-        consumer="apm"
-        alertTypeId={alertType}
-        canChangeTrigger={false}
-      />
-    )
+  const {
+    services: { triggersActionsUi },
+  } = useKibana<KibanaDeps>();
+  const AddAlertFlyout = useMemo(
+    () =>
+      alertType &&
+      triggersActionsUi.getAddAlertFlyout({
+        consumer: 'apm',
+        addFlyoutVisible,
+        setAddFlyoutVisibility,
+        alertTypeId: alertType,
+        canChangeTrigger: false,
+      }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [addFlyoutVisible, alertType]
   );
+  return <>{AddAlertFlyout}</>;
```

Comment:
Is there a reason why this cannot be a component that the alerting UI exports?

Reply:
From the code reducing prospective it sounds right to expose it as component, but our decision was to live the details of the flyout usage to the end plugin - as a result it can be without memoizations or with the own dependancies list.

Reply:
So the memoization is not needed?

Reply:
This looks ok, but since it's not a component could we make the variable lowercase? Also why are we disabling exhaustive-deps? If there's a good reason for that please add a comment explaining why.